internal/controlplane/handlers_ruletype.go

```diff
@@ -373,3 +377,48 @@ func validateMarkdown(md string) error {
 
 	return nil
 }
+
+func (s *Server) validateDataSources(
+	ctx context.Context,
+	projectID uuid.UUID,
+	ruleDS []*minderv1.DataSourceReference,
+) error {
+	// Short circuiting to avoid accessing the database.
+	if len(ruleDS) == 0 {
+		return nil
+	}
+
+	if len(ruleDS) > 0 && !flags.Bool(ctx, s.featureFlags, flags.DataSources) {
+		return status.Errorf(codes.Unavailable, "DataSources feature is disabled")
+	}
+
+	requested := make([]string, 0)
+	for _, ds := range ruleDS {
+		requested = append(requested, ds.Name)
+	}
+
+	// List Data Sources defined in the whole project hierarchy.
+	availableDS, err := s.dataSourcesService.List(
```

Comment:
There could be many data sources and only one or two references by a rule type. Wouldn't it make more sense to do it the other way around and query only the ones being instantiated by the rule?

Reply:
It's hard to tell whether this is going to be a performance hit without some data, but the trade off is
* current implementation: potentially many rule types, but only one round trip with the database
* your suggestion: fewer data loaded at rule creation time, but potentially more round trips to the database

Given it's reasonable to think that rules will use few data sources rather than many, I'll change this to load only the necessary ones, but I won't add a new statement for bulk-loading and will use [`GetDataSourceByName`](https://github.com/mindersec/minder/blob/main/database/query/datasources.sql#L55-L57) instead.